src/Drupal/Commands/core/LocaleCommands.php

```diff
@@ -128,4 +131,89 @@ public function update($options = ['langcodes' => self::REQ])
 
         drush_backend_batch_process();
     }
+
+    /**
+     * Exports to a gettext translation file.
+     *
+     * @command locale:export
+     * @drupal-dependencies locale
+     * @option langcode The language code of the exported translations.
+     * @option types String types to include, defaults to all types.
+     *      Types: 'not-customized', 'customized', 'not-translated'.
+     * @usage drush locale:export --langcode=nl > nl.po
+     *   Export the Dutch non-customized translations.
+     * @usage drush locale:export --langcode=nl --types=customized,not-customized > nl.po
+     *   Export the Dutch customized and not customized translations.
+     * @usage drush locale:export > drupal.pot
+     *   Export the template file to translate.
+     * @aliases locale-export
+     * @see \Drupal\locale\Form\ExportForm::submitForm
+     *
+     * @throws \Exception
+     */
+    public function export($options = ['langcode' => self::OPT, 'types' => self::OPT])
+    {
+        // If template is required, language code is not given.
+        if ($options['langcode'] != LanguageInterface::LANGCODE_SYSTEM) {
+            $language = \Drupal::languageManager()->getLanguage($options['langcode']);
+
+            if ($language == null) {
+                $this->logger()->warning(dt('Language code @langcode is not configured.', [
+                    '@langcode' => $options['langcode'],
+                ]));
+                return;
+            }
+        } else {
+            $language = null;
+        }
+
+        $reader = new PoDatabaseReader();
+        $language_name = '';
+
+        if ($language != null) {
+            $options['types'] = explode(',', $options['types']);
+            $allowed_types = [
+                'not-customized',
+                'customized',
+                'not-translated',
+            ];
+
+            if (empty(reset($options['types']))) {
```

Comment:
That was my first idea also, but by default we have $options[...][0] = 0 this is because of the default state of 'types' with only --types you'll get $options['types'][0] = 1;